pytorch_lightning/profiler/profilers.py

```diff
@@ -88,15 +117,41 @@ def profile_iterable(self, iterable, action_name: str) -> None:
 
     def describe(self) -> None:
         """Logs a profile report after the conclusion of the training run."""
+        self._prepare_file()
         for write in self.write_streams:
             write(self.summary())
+        if self.output_file:
+            self.output_file.flush()
+        self.teardown()
```

Comment:
`teardown` could be called directly by the trainer during the teardown hooks. do we want to nest it inside of `describe` ?

Reply:
@tchaton I believe you added it. Any reason to have it here? We already have it in:

https://github.com/PyTorchLightning/pytorch-lightning/blob/01a760e7a42ae0686a67ca80b5f06188e5f7f9f7/pytorch_lightning/trainer/trainer.py#L1077-L1085

Reply:
Hey,

If a user was to perform a deepcopy in a hook between the describe call and the teardown call, 
it would fail in python 3.6 as write_streams can't be pickled.

IMO, I think it is safer to close the write_streams as soon as being used.

Reply:
After discussing, we chose to keep it as is. Will add a comment mentioning the pickle limitation.